text/0000-rustdoc-json.md

```diff
@@ -0,0 +1,1124 @@
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this associated const.
+`default`     | String | (*Optional*) The stringified expression for the default value, if provided.
+
+### `kind == "assoc_type"`
+
+These items only show up in trait _definitions_. When looking at a trait impl item, the item where the associated type is defined is a `"typedef"` item.
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`bounds`      | [[GenericBound](#GenericBound)] | The bounds for this associated type.
+`default`     | [Type](#Type) | (*Optional*) The default for this type, if provided.
+
+### `kind == "impl"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`is_unsafe`   | bool     | Whether this impl is for an unsafe trait.
+`generics`    | [Generics](#Generics) | Information about the impl's type parameters and `where` clauses.
+`provided_trait_methods` | [String] | The list of names for all provided methods in this impl block. This is provided for ease of access if you don't need more information from the `items` field.
+`trait`       | [Type](#Type) | The trait being implemented or `null` if the impl is "inherent".
+`for`         | [Type](#Type) | The type that the impl block is for.
+`items`       | [[ID](#ID)] | The list of method, constant, and typedef items contained in this impl block.
+`negative`    | bool     | Whether this is a negative impl (e.g. `!Sized` or `!Send`).
+`synthetic`   | bool     | Whether this is an impl that's implied by the compiler (for autotraits).
+`blanket_impl` | String | (*Optional*) The name of the generic parameter used for the blanket impl, if this impl was produced by one. For example `impl<T, U> Into<U> for T` would result in `blanket_impl == "T"`.
+
+### `kind == "constant"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this constant.
+`expr`        | String   | The stringified expression of this constant.
+`value`       | String   | (*Optional*) The value of the evaluated expression for this constant, which is only computed for numeric types.
+`is_literal`  | bool     | Whether this constant is a bool, numeric, string, or char literal.
+
+### `kind == "static"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this static.
+`expr`        | String   | The stringified expression that this static is assigned to.
```

Comment:
This seems like a weird thing to expose in the public API (and is also very underspecified). What's the rationale for including it?

Reply:
Similar to constants, you might want to show the value a static is assigned to:

```rust
pub static controller: Controller = Controller::new("some publicly useful arg", 42);
```

Reply:
This seems like it's mostly for pretty printing, so I think it should be marked as unstable and subject to change. Rustc has no stability guarantee on `-Z unpretty=*` anywhere else, I don't think we should add it here.